proto/distancesensorservice/v1/distancesensorservice.proto

```diff
@@ -0,0 +1,95 @@
+// MIT License
+// 
+// Copyright (c) 2021 Corvin Lasogga
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+syntax = "proto3";
+
+package distancesensorservice.v1;
+
+option go_package = "coe-sops/proto/distancesensorservice/distancesensorservice_v1";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+
+service DistanceSensorController{
+    // Returns the measurement range according to the sensor specification
+    // of the manufacturer.
+    rpc GetMeasuringRange(google.protobuf.Empty) returns (MeasuringRange);
+    // Measures the distance with the required parameters.
+    rpc MeasureDistance(MeasuringRequest) returns (MeasuringResponse);
+}
+
+// Start and end of the measuring range according to the sensor specification
+// of the manufacturer.
+message MeasuringRange {
+    double start_distance_in_mm = 1;
+    double end_distance_in_mm = 2;
+}
+
+// Measuring request with the measuring rate in kHz.
+message MeasuringRequest{
+    float measuring_rate_khz = 1;
```

Comment:
Is the frequency only relevant for the Measure with trigger? Maybe add a comment to clarify

Reply:
At the moment: It is an input for both, a single measurement as well as for the triggered measurement.

Generally speaking: The lower the measurement frequency, normally the higher the exposure time and such a more accurate measurement is possible. Furthermore, the measurement at the surface with a higher angle is possible.